rwengine/src/objects/VehicleObject.cpp

```diff
@@ -351,7 +351,7 @@ void VehicleObject::tickPhysics(float dt) {
                     if (_lastHeight >= wH) {
                         // we are for real, underwater
                         inWater = true;
-                    } else if (inWater == false) {
+                    } else if (inWater == false) { // This condition is unnecessary
```

Comment:
The code itself is pretty broken and does not what the comment suggests. The check for line 350 is seemingly never done. We also had issue #214 which is probably affected by this.
So it probably needs a full refactor.
However, seeing how this is mostly a small cleanup, I'd suggest linking this "weird" code from #214 and not worrying about it now (e.g. remove this part from the PR).